library/std/src/os/vita/raw.rs

```diff
@@ -34,37 +31,3 @@ pub type off_t = libc::off_t;
 
 #[stable(feature = "raw_ext", since = "1.1.0")]
 pub type time_t = libc::time_t;
-
-#[repr(C)]
-#[derive(Clone)]
-#[stable(feature = "raw_ext", since = "1.1.0")]
-pub struct stat {
-    #[stable(feature = "raw_ext", since = "1.1.0")]
-    pub st_dev: dev_t,
-    #[stable(feature = "raw_ext", since = "1.1.0")]
-    pub st_ino: ino_t,
-    #[stable(feature = "raw_ext", since = "1.1.0")]
-    pub st_mode: mode_t,
-    #[stable(feature = "raw_ext", since = "1.1.0")]
-    pub st_nlink: nlink_t,
-    #[stable(feature = "raw_ext", since = "1.1.0")]
-    pub st_uid: uid_t,
-    #[stable(feature = "raw_ext", since = "1.1.0")]
-    pub st_gid: gid_t,
-    #[stable(feature = "raw_ext", since = "1.1.0")]
-    pub st_rdev: dev_t,
-    #[stable(feature = "raw_ext", since = "1.1.0")]
-    pub st_size: off_t,
-    #[stable(feature = "raw_ext", since = "1.1.0")]
-    pub st_atime: time_t,
-    #[stable(feature = "raw_ext", since = "1.1.0")]
-    pub st_mtime: time_t,
-    #[stable(feature = "raw_ext", since = "1.1.0")]
-    pub st_ctime: time_t,
-    #[stable(feature = "raw_ext", since = "1.1.0")]
-    pub st_blksize: blksize_t,
-    #[stable(feature = "raw_ext", since = "1.1.0")]
-    pub st_blocks: blkcnt_t,
-    #[stable(feature = "raw_ext", since = "1.1.0")]
-    pub st_spare4: [c_long; 2usize],
-}
```

Comment:
Huh, I'm surprised removing this builds, since what I previously assumed was that the entire reason this module exists at all is because even though we deprecated it and thus do not add it for new OS, we export some things via `std::os::unix::raw` and `cfg` based on `target_family = "unix"` for that, thus anything that claims to be a "Unix" requires this.

I assume this was triggering dead code warnings?

Reply:
> I assume this was triggering dead code warnings?

Yep. This struct exists in libc, which is the exact reason why [`std::os::linux::raw::stat`](https://doc.rust-lang.org/nightly/std/os/linux/raw/struct.stat.html) is deprecated. So given the dead code warning I removed this definition.

Reply:
> I previously assumed was that the entire reason this module exists at all is because even though we deprecated it and thus do not add it for new OS, we export some things via `std::os::unix::raw` and `cfg` based on `target_family = "unix"` for that, thus anything that claims to be a "Unix" requires this.

Hmmm yeah, for the other types that sounds right, but [given that `stat` is not re-exported via `std::os::unix::raw`](https://github.com/rust-lang/rust/blob/master/library/std/src/os/unix/raw.rs), I guess that there was something internal to `std` still using `platform::raw::stat` by the time `vita` was added? 

Also, the same thing happened to 3DS: https://github.com/rust-lang/rust/pull/127996#discussion_r1685182103

Reply:
That or the dead code lints were more imprecise.